packages/image/README.md

```diff
@@ -0,0 +1,132 @@
+# `@sveltejs/image`
+
+**WARNING**: This package is experimental. It uses pre-1.0 versioning and may introduce breaking changes with every minor version release.
+
+This package aims to bring a plug and play image component to SvelteKit that is opinionated enough so you don't have to worry about the details, yet flexible enough for more advanced use cases or tweaks. It serves a smaller file format like `avif` or `webp` and uses the `srcset` and `sizes` attributes of the `img` tag to provide resized images suitable for various device sizes, which for example results in smaller images downloaded for mobile.
+
+## Setup
+
+Install:
+
+```bash
+npm install --save @sveltejs/image
+```
+
+Adjust `vite.config.js`:
+
+```diff
++import { images } from '@sveltejs/image/vite';
+import { sveltekit } from '@sveltejs/kit/vite';
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+	plugins: [
++		images({ providers: { default: '@sveltejs/image/providers/<choose one>' } }),
+		sveltekit()
+	]
+});
+```
+
+> `<choose one>` refers to choosing one of the ready-to-use providers. We plan to add more providers over time. You can create your own by creating a JavaScript file with a `export function getURL({ src, width, height }): string` function inside and then pointing to that file in the config.
+
+In case of Vercel, also adjust your `svelte.config.js`:
+
+```diff
+import adapter from '@sveltejs/adapter-vercel';
+
+/** @type {import('@sveltejs/kit').Config} */
+const config = {
+	kit: {
+-		adapter: adapter()
++		adapter: adapter({ images: true })
+	}
+};
+
+export default config;
+```
+
+## Usage
+
+### Dynamic runtime optimization using one of the providers, i.e. an image CDN:
+
+```svelte
+<script>
+	import Image from '@sveltejs/image';
+</script>
+
+<Image src="/path/to/your/image.jpg" width={1200} height={1800} alt="An alt text" />
+```
+
+`width` and `height` should be the intrinsic width/height of the referenced image - i.e. the dimensions of the image before styling. `alt` should describe the image. All are required. The `src` is transformed by calling `getURL` of the `default` provider provided in the `vite.config.js`.
+
+If you want to optimize an image from an external URL, add its domain to the `domains` config of the plugin.
+
+### Static build time optimization:
+
+Static build time optimization uses `vite-imagetools`, which comes as an optional peer dependency, so you first need to install it:
```

Comment:
I think we probably need some docs in this section about how installing `vite-imagetools` will change the behaviour of existing imports. In my app I had this...

```js
import logo from '$lib/icons/logo.avif';
```

```svelte
<img alt="logo" src={logo} />
```

...and it broke. (Ideally it wouldn't break in the first place, i.e. you'd have to explicitly opt in to `vite-imagetools` handling an import, but I'm not sure that's possible?)

Reply:
It would be possible if we expose our own imagetools directive, so people would do something like `import Foo from './my/image.jpg?as=svelteimage'` if they want to import it manually, and the preprocessor would add that automatically